vscode/src/data/storage.ts

```diff
@@ -12,30 +12,22 @@ import {
 
 const MAX_FILES = 5;
 
-const buildDataFolderPath = () => {
+// This function returns the data folder, creating it if it doesn't exist
+// and returning undefined when no workspace is loaded.
+export const getDataFolder = () => {
   const firstWorkspace = vscode.workspace.workspaceFolders?.[0];
   if (!firstWorkspace) {
+    console.log("Cannot create data folder due to no workspace");
     return;
   }
 
-  return path.join(firstWorkspace.uri.fsPath, ".vscode", "konveyor");
-};
-
-const getDataFolder = () => {
-  const dataFolderPath = buildDataFolderPath();
-  return dataFolderPath && fs.existsSync(dataFolderPath) ? dataFolderPath : undefined;
-};
+  const dataFolderPath = path.join(firstWorkspace.uri.fsPath, ".vscode", "konveyor");
 
-const createDataFolderIfNeeded = async () => {
-  if (getDataFolder()) {
-    return;
-  }
-  const dataFolderPath = buildDataFolderPath();
-  if (dataFolderPath) {
-    await vscode.workspace.fs.createDirectory(vscode.Uri.file(dataFolderPath));
-  } else {
-    console.log("Cannot create data folder due to no workspace");
+  if (fs.existsSync(dataFolderPath)) {
+    return dataFolderPath;
   }
+  vscode.workspace.fs.createDirectory(vscode.Uri.file(dataFolderPath));
```

Comment:
This is an excellent explanation of the situation. Thank you.

Maybe, instead of this mess I created, I update `createDataFolderIfNeeded` to return the path to the folder it created (and export that)?